### PR TITLE
Adds some minor polish for explorers

### PR DESCRIPTION
**Pull request by orta**
Started reading the code, added a few bits of docs and notes

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -1,6 +1,29 @@
...
+if (errors.length) {
+    title("> Errors:")
+    console.log(errors)
+}
+
+title("> Output:")
+console.log(js)
+
+// Print errors, write js to file
+fs.writeFileSync(args[0] + '.js', js)
```
Comment (orta) [inline index.ts:29]: Basically shows you the before after etc. Given that this isn't really for transpiling, I think showing all the interesting stuff form the CLI makes sense

Screen Shot 2021-06-07 at 6 07 51 PM

File: package.json [hunk 2]
```diff
@@ -2,12 +2,13 @@
...
-  "main": "index.js",
+  "main": "compiler/index.js",
   "scripts": {
     "build": "tsc",
     "test": "rm baselines/local/*; tsc && node test.js",
     "accept": "mv baselines/local/* baselines/reference/",
-    "run": "node index.js"
+    "run": "node compiler/index.js",
+    "mtsc": "node compiler/index.js"
   },
```
Comment (orta) [inline package.json:12]: `npm run run` haha, so I gave it a different name

File: tsconfig.json [hunk 3]
```diff
@@ -3,7 +3,8 @@
         "module": "commonjs",
         "strict": true,
         "target": "esnext",
-        "lib": ["es2020", "dom"]
+        "lib": ["es2020", "dom"],
+        "outDir": "compiler"
```
Comment (orta) [inline tsconfig.json:7]: Moved JS into another dir - having the JS in the same folder makes it harder to browse

Comment (DanielRosenwasser) [reply tsconfig.json:7]: I would even suggest moving all the source and `tsconfig.json` into a `src` directory - my personal take is that it's hard to discern what I should be looking for.

Comment (orta) [reply tsconfig.json:7]: I don't disagree with ^ either

File: README.md [hunk 4]
```diff
@@ -5,6 +5,24 @@ I started this project as part of reading [Modern Compiler Implementation in ML]
...
+```sh
+git clone https://github.com/sandersn/mini-typescript
+cd mini-typescript
+code .
+
+# Get set up
+npm i
+npm run build
+
+# Rr have your changes instantly show
```
Comment (sandersn) [inline README.md:19]: Not sure what `Rr` is, so..

```suggestion
# Have your changes instantly show
```

File: package.json (hunk 2 above)
Comment (sandersn) [reply package.json:12]: Might as well delete `run` then, I copied that early on from the F# version and never deleted it.

File: package.json [hunk 5]
```diff
@@ -2,12 +2,13 @@
   "name": "mini-typescript",
   "version": "1.0.0",
   "description": "A miniature model of the Typescript compiler",
-  "main": "index.js",
+  "main": "compiler/index.js",
   "scripts": {
     "build": "tsc",
     "test": "rm baselines/local/*; tsc && node test.js",
```
Comment (sandersn) [inline package.json:8]:
```suggestion
    "test": "rm baselines/local/*; tsc && node compiler/test.js",
```

File: index.ts (hunk 1 above)
Comment (sandersn) [reply index.ts:29]: Thanks for adding all this. (Until yesterday, this file just had `console.log("I am compiler")` in it.)

File: tsconfig.json (hunk 3 above)
Comment (sandersn) [reply tsconfig.json:7]: Do tests work with this? They use a relative path to reference baselines/local/... and baselines/reference/... I don't know whether that will end up relative to the root where it's run or relative to compiler/test.js.

**Review by sandersn — CHANGES_REQUESTED** [review]
I like your changes, although I kind of want to set up CI before accepting PRs. It might be a day or so.

File: tsconfig.json (hunk 3 above)
Comment (sandersn) [reply tsconfig.json:7]: I set up github actions so ... let's see what happens when I close/re-open (I can't remember if you have to merge from main)

File: package.json [hunk 6]
```diff
@@ -2,12 +2,13 @@
   "name": "mini-typescript",
   "version": "1.0.0",
   "description": "A miniature model of the Typescript compiler",
-  "main": "index.js",
+  "main": "compiler/index.js",
```
Comment (sandersn) [inline package.json:5]: I try to use compiler names for things when it's not wordy or confusing. I vote for `built`:
```suggestion
  "main": "built/index.js",
```

Comment (orta) [reply package.json:5]: Cool, agree

**Comment by orta** [comment]
Took some time to look at the testing situation too, I had some legit fails - so I made the output explain what's wrong and copied Jest's `--write` param to update baselines

Screen Shot 2021-06-08 at 4 15 21 PM

File: baselines/reference/singleVar.ts.errors.baseline [hunk 7]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (sandersn) [inline baselines/reference/singleVar.ts.errors.baseline:1]: these baselines are the same as singleVar.*.baseline except they have an additional .ts in the filename

**Review by sandersn — CHANGES_REQUESTED** [review]
The test changes are not removing the `.ts` extension.

It would probably be good if the mtsc entrypoint also removed the `.ts` extension actually.

**Review by sandersn — APPROVED** [review]

**Pull request merged**